aten/src/TH/THStorage.cpp

```diff
@@ -96,3 +96,145 @@ THStorage* THStorage_newWithAllocator(at::ScalarType scalar_type, ptrdiff_t size
   storage->device = INT_MIN;  // device is not meaningful on CPU
   return storage;
 }
+
+ptrdiff_t THStorage_size(const THStorage *self)
+{
+  return self->size;
+}
+
+size_t THStorage_elementSize(const THStorage *self)
+{
+  return at::elementSize(self->scalar_type);
+}
+
+THStorage* THStorage_newWithMapping(at::ScalarType scalar_type, const char *filename, ptrdiff_t size, int flags)
+{
+  THMapAllocatorContext *ctx = THMapAllocatorContext_new(filename, flags);
+
+  THStorage *storage = THStorage_newWithAllocator(scalar_type, size,
+                                                  &THMapAllocator,
+                                                  ctx);
+
+  if(size <= 0)
+    storage->size = THMapAllocatorContext_size(ctx)/THStorage_elementSize(storage);
+
+  THStorage_clearFlag(storage, TH_STORAGE_RESIZABLE);
+
+  return storage;
+}
+
+void THStorage_setFlag(THStorage *storage, const char flag)
+{
+  storage->flag |= flag;
+}
+
+void THStorage_clearFlag(THStorage *storage, const char flag)
+{
+  storage->flag &= ~flag;
+}
+
+void THStorage_retain(THStorage *storage)
+{
+  if(storage && (storage->flag & TH_STORAGE_REFCOUNTED))
+    ++storage->refcount;
```

Comment:
nit: when I've done these moves, I've added braces in this situation to match the way most of our ATen/pytorch C++ code is written.

Reply:
easy enough to fix